### PR TITLE
Eliminate plugin system

### PR DESCRIPTION
**Pull request by hypesystem**
This PR step-by-step eliminates anything to do with the "plugin"-system.

Plugins do a few things: they can have an initialization function, register routes in the express app, and reference some module.

After this PR we just call or inline the initialization functions that exist at server startup, register routes as directly as possible, and use requires instead of getting plugins via the plugin controller.

I have tried to keep the changes in manageable steps.

One thing of note: when I pruned packages with a new version of npm it added *a lot* of lines to the lock file. I don't really know what that's about.

I have tried to test everything and it seems to work for me - if you can get it to work + think the changes make sense, let's get this into beta for more testing :smile:

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -17,64 +16,48 @@ const co = {
...
-            throw new Error('Could not start server, needed "port" and "ip" or "socketPath" in the configuration.');
-          }
-        }, (err) => {
-          console.error('Error when starting the app: ', err.stack);
-          process.exit(2);
-        });
-    });
+    } else if(config.socketPath) {
+      app.listen(config.socketPath, function() {
+        console.log('Express server listening on socket %s, in %s mode', config.socketPath, app.get('env'));
```
Comment (ath88) [inline server.js:69]: We don't support the socket-method anymore - so it can be removed. It was overly cautious, and not useful since we removed nginx.

**Review by ath88 — APPROVED** [review]
Amazing stuff! I love your commit messages. It would be nice to have a separate subject line and body though. Take a look at https://chris.beams.io/posts/git-commit

**Comment by ath88** [comment]
You could rebase the entire thing on the newest master - it includes building via Github Action - which I predict will fail, since you removed a file used in the dockerfile.

File: plugins/auth.js [hunk 2]
```diff
@@ -27,7 +25,6 @@ module.exports = {
 
     app.use(function(req, res, next) {
       res.locals.user = req.user;
-      currentUser = req.user;
```
[the remark this answers is not in the document]
Comment (hypesystem) [reply plugins/auth.js:30]: Yeah these were the only references to the variable in the file, don't know why it was there. It isn't passed anywhere or exposed in any way.

[the remark this answers is not in the document]
Comment (hypesystem) [reply plugins/auth.js:30]: Also if it was used it would significantly break things, so good thing it wasn't I guess?

File: server.js (hunk 1 above)
Comment (hypesystem) [reply server.js:69]: Ok :smile: Let's do it in the next PR though.

**Comment by hypesystem** [comment]
@ath88 ligner at jeg har brugt en nyere npm-version end den der bruges i docker imaget til `npm i`: https://github.com/CopenhagenCityArchives/kbh-billeder/pull/294/checks?check_run_id=3713639125#step:6:248

Hvad er den bedste måde at opdatere npm på?

**Pull request merged**